### PR TITLE
feat(journal): add extra search parameters

### LINKED ISSUE #2547: Search Journal on Record, Entity, and Comments
The Journal is missing three crucial search fields:
 1. Record Identifier (`VO.TPA.1`, `IV.TPA.34`, `CP.TPB.1`, etc)
 2. Entity Identifier (`PA.HEV.1`, etc)
 3. The comments field.

All these should be added to increase the usability of the system.

### PR DESCRIPTION
**Pull request by jniles**
This commit adds the following search parameters to the Posting Journal:
 1. `comment` - fuzzy searches the "comment" column
 2. `hrRecord` - searches the "record" column
 3. `hrEntity` - searches the "entity" column
 4. `hrReference` - searches the "reference" column

Closes #2547.

### CODE REVIEW

File: client/src/modules/journal/modals/search.modal.html [hunk 1]
```diff
@@ -96,6 +123,21 @@
           </div>
         </div>
 
+        <!-- description fuzzy search -->
```
Comment (mbayopanda) [inline client/src/modules/journal/modals/search.modal.html:126]: That is a very cool feature 👍

**Review by mbayopanda — APPROVED** [review]
LGTM

**Comment by jniles** [comment]
Hmm... looks like we have some flaky tests.

**Pull request merged**